### PR TITLE
Fix helpers arch support for upgrade tests

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Fix helpers arch support for upgrade tests, previously url's for installing previous were hard-coded to use Intel, x86_64 and amd64.  The fix checks if AARCH64 and sets the architecture to use appropriately

### What issues does this PR fix or reference?
Fixes: n/a

### New Behavior
Now test upgrade on ARM64

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tools/ci.py [hunk 1]
```diff
@@ -476,7 +476,11 @@ def matrix(ctx: Context, distro_slug: str):
     _matrix = []
     for transport in ("zeromq", "tcp"):
         if transport == "tcp":
-            if distro_slug not in ("centosstream-9", "ubuntu-22.04-arm64"):
+            if distro_slug not in (
```
Comment (Ch3LL) [inline tools/ci.py:479]: We need to remove all of the tools changes. They are editing the jobs for the main test suite.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: pkg/tests/support/helpers.py [hunk 2]
```diff
@@ -561,20 +565,29 @@ def install_previous(self):
...
             ret = self.proc.run(self.pkg_mngr, "install", "apt-transport-https", "-y")
             self._check_retcode(ret)
+            ## only classic 3005 has arm64 support
+            if self.major >= "3006" and platform.is_aarch64():
+                arch = "arm64"
+            elif platform.is_aarch64() and self.classic:
+                arch = "arm64"
+            else:
+                arch = "amd64"
+            ret = self.proc.run("mkdir", "-p", "/etc/apt/keyrings")
```
[the remark this answers is not in the document]
Comment (dmurphy18) [reply pkg/tests/support/helpers.py:575]: no problem, will do, was doing quick debugging and missed it in the clean up

File: tools/ci.py (hunk 1 above)
Comment (dmurphy18) [reply tools/ci.py:479]: If you recall, I asked you about this, and you replied yes, ubuntu-22.04 needed to be in there too.

Hence not understanding why you want to pull this unless in its own PR ?

File: pkg/tests/support/helpers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmurphy18) [reply pkg/tests/support/helpers.py]: Done, please review change

**Review by Ch3LL — APPROVED** [review]

**Review by Ch3LL — APPROVED** [review]

File: pkg/tests/support/helpers.py [hunk 3]
```diff
@@ -561,20 +565,28 @@ def install_previous(self):
...
             ret = self.proc.run(self.pkg_mngr, "install", "apt-transport-https", "-y")
             self._check_retcode(ret)
+            ## only classic 3005 has arm64 support
+            if self.major >= "3006" and platform.is_aarch64():
+                arch = "arm64"
+            elif platform.is_aarch64() and self.classic:
+                arch = "arm64"
+            else:
+                arch = "amd64"
+            pathlib.PATH("/etc/apt/keyrings").mkdir(parents=True, exist_ok=True)
```
Comment (Ch3LL) [inline pkg/tests/support/helpers.py:575]:
```suggestion
            pathlib.Path("/etc/apt/keyrings").mkdir(parents=True, exist_ok=True)
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: pkg/tests/support/helpers.py (hunk 3 above)
Comment (dmurphy18) [reply pkg/tests/support/helpers.py]: Sorry about the fat finger

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**